plugins/lookup/dig.py

```diff
@@ -300,6 +308,11 @@ def run(self, terms, variables=None, **kwargs):
                         rdclass = dns.rdataclass.from_text(arg)
                     except Exception as e:
                         raise AnsibleError("dns lookup illegal CLASS: %s" % to_native(e))
+                elif opt == 'retry_servfail':
+                    try:
+                        myres.retry_servfail = bool(arg)
+                    except Exception as e:
```

Comment:
I don't think setting this attribute will ever raise an exception. (See [here](https://dnspython.readthedocs.io/en/stable/resolver-class.html#dns.resolver.Resolver.retry_servfail))

Reply:
Removed try/except block in 2db16c360cc1dbb42077cac4bcc04891b538e66e.